File: src/main/java/com/github/bgora/american/AmericanCarFactory.java
package com.github.bgora.american;

import java.util.List;
import java.util.stream.Collectors;

public class AmericanCarFactory {

    public static AmericanCar createCar(final String name, final Integer horsePower, final Integer maxSpeed) {
        return new AmericanCar(name, horsePower, maxSpeed);
    }

    public static List<AmericanCar> produce(List<String> cars) {
        return cars.stream().map(car -> {
            var attributes = car.split(";");
            return createCar(attributes[0], Integer.parseInt(attributes[1]), Integer.parseInt(attributes[2]));
        }).collect(Collectors.toList());
    }
}

File: src/main/kotlin/com/github/bgora/american/AmericanCarFactory.kt
package com.github.bgora.american

import java.util.stream.Collectors

object AmericanCarFactory {

    fun createCar(name: String, horsePower: Int, maxSpeed: Int): AmericanCar {
        return AmericanCar(name, horsePower, maxSpeed)
    }

    fun produce(cars: List<String>): List<AmericanCar> {
        return cars.stream().map { car: String ->
            val attributes = car.split(";".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
            createCar(attributes[0], attributes[1].toInt(), attributes[2].toInt())
        }.collect(Collectors.toList())
    }
}